doc/developers/maintainer.rst.template

```diff
@@ -0,0 +1,385 @@
+        {%- endif %}
+        * [ ] Set the version number in the release branch
+        * [ ] Check that the wheels for the release can be built successfully
+        * [ ] Merge the PR with `[cd build]` commit message to upload wheels to the staging repo
+        * [ ] Upload the wheels and source tarball to https://test.pypi.org
+        * [ ] Create tag on the main repo
+        * [ ] Confirm bot detected at https://github.com/conda-forge/scikit-learn-feedstock
+              and wait for merge
+        * [ ] Upload the wheels and source tarball to PyPI
+        {%- if key != "rc" %}
+        * [ ] Publish to https://github.com/scikit-learn/scikit-learn/releases
+        {%- endif %}
+        * [ ] Announce on mailing list and on Twitter, and LinkedIn
+        {%- if key == "final" %}
+        * [ ] Update symlink for stable in https://github.com/scikit-learn/scikit-learn.github.io
+        {%- endif %}
+        {%- if key != "rc" %}
+        * [ ] Update SECURITY.md in main branch
+        {%- endif %}
+
+    {% if key != "rc" %}
+    - Rebase this PR from the `{{ version_main }}.X` branch:
+
+      .. prompt:: bash
+
+        git rebase -i upstream/{{ version_main }}.X
+
+      This will open an interactive rebase with the `git-rebase-todo` containing all the
+      latest commits on `main`. At this stage, you have to perform this interactive
+      rebase with at least someone else (to not forget something and to avoid doubts).
+
+      - Do not remove lines but drop commit by replacing `pick` with `drop`.
+      - Commits to pick for a bug-fix release are *generally* prefixed with `FIX`, `CI`,
+        and `DOC`. They should at least include all the commits of the merged PRs that
+        were milestoned for this release and/or documented as such in the changelog.
+      - Commits to `drop` for a bug-fix release are *generally* prefixed with `FEAT`,
+        `MAINT`, `ENH`, and `API`. Reasons for not including them is to prevent change
+        of behavior (which should only happen in major/minor releases).
+      - After having dropped or picked commits, **do not exit** but paste the content of
+        the `git-rebase-todo` message in the PR. This file is located at
+        `.git/rebase-merge/git-rebase-todo`.
+      - Save and exit to start the interactive rebase. Resolve merge conflicts when
+        necessary.
+    {% endif %}
+
+    {% if key == "rc" %}
+    - Create a PR from `main` and targeting `main` to increment the `__version__`
+      variable in `sklearn/__init__.py` to increment the dev0 version. This means while
+      we are in the release candidate period, the latest stable is two version behind
+      the `main` branch, instead of one. In this PR targeting `main`, you should also
+      include a new what's new file under the `doc/whats_new/` directory so PRs that
+      target the next version can contribute their changelog entries to this file in
+      parallel to the release process.
```

Comment:
```suggestion
    - Create a PR from `main` and targeting `main` to increment the dev0 `__version__`
      variable in `sklearn/__init__.py`. This means while we are in the release
      candidate period, the latest stable is two version behind the `main` branch,
      instead of one. In this PR targeting `main`, you should also include a new what's
      new file under the `doc/whats_new/` directory so PRs that target the next version
      can contribute their changelog entries to this file in parallel to the release
      process.
```